Update pact-go v1 to use latest pact-ruby-standalone (ruby 3.2 + arm64 support)

YOU54F opened the pull request:
as per #290 but retargeted against v1.x.x branch

_____

## Changes to traveling ruby to support additional platforms

### Details of changes made to travelling ruby

PR's to Travelling ruby to update the supported archs

- https://github.com/phusion/traveling-ruby/pull/132
- https://github.com/phusion/traveling-ruby/pull/133

Which means we can now support


| OS     | Ruby      | Architecture | Supported |
| -------| ------- | ------------ | --------- |
| OSX    | 3.2.2     | x86_64       | ✅         |
| OSX    | 3.2.2     | aarch64 (arm)| ✅         |
| Linux  | 3.2.2   | x86_64       | ✅         |
| Linux  | 3.2.2   | aarch64 (arm)| ✅          |
| Windows| 3.2.2 | x86_64       | ✅        |
| Windows| 3.2.2 | x86       | ✅        |
| Windows| 3.2.2 | aarch64 (via x86 emulation) |  ✅        |

### Details of changes made to pact-ruby-standalone

- https://github.com/pact-foundation/pact-ruby-standalone/pull/100
- https://github.com/pact-foundation/pact-ruby-standalone/releases/tag/v2.0.1

### Details of changes made to pact-go

These have been consumed in the pact-go project, and tested cross platform

In order to test ARM64 on MacOS, and Linux, I have utilised cirrus-ci, which is free for open-source usage (within reasonable limits)

| OS     | Standalone Version     | Architecture | Tested |
| -------| ------- | ------------ | --------- |
| OSX    | v2.0.3    | x86_64       | GitHub Actions            |
| OSX    | v2.0.3     | aarch64 (arm)| Cirrus CI         |
| Linux  | v2.0.3  | x86_64       | GitHub Actions          |
| Linux  | v2.0.3  | aarch64 (arm)| Cirrus-CI       |
| Windows| v2.0.3 | x86_64       | untested        |
| Windows| v2.0.3 | aarch64 (via x86 emulation) |  untested    |
| Windows| v2.0.3 | x86       | untested      |

### User facing notes

-  Linux
  - Adds support for arm64
  - Drops support for x86 Linux
- Adds support for macos arm64
  - macOS arm64 (m1/m2) users will no require require installing rosetta
- Windows
  - i386 and x86_64 builds available

mefellows commented:
Nice! Thanks.

Looking at the commits, nothing would appear in the changelog. Is there a way to rebase and update a commit to add a `feat`?

In this package, I usually go for a `feat: update to standalone x.y.z`

YOU54F commented:
np makes total sense, have done that but just yak shaving some things whilst I am here - will poss split these out into sep PR's

- golanglint-ci job failing, post v3 it requires a setup-go step
  - https://github.com/golangci/golangci-lint-action#compatibility
  - it fails on 1.18.x go version, not sure if we should run against both
- Appveyor job hasn't passed in a loooooooong time (2-3 years)
  - Happy to look at it, might be failing to connect to docker, as windows (https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host)
- Updated Pact-broker config to use pactfoundation image over dius, and added healthcheck
